scripts/monitor_synced.py

```diff
@@ -0,0 +1,128 @@
+    for i in range(NUM_RETRIES):
+        cmd = ['curl', '--location', '--request', 'POST', f'https://explorer-rpc.{env}.sui.io/',
+               '--header', 'Content-Type: application/json', '--data-raw',
+               '{"jsonrpc":"2.0", "method":"suix_getEpochs", "params":[null, 1, true], "id":1}']
+        try:
+            result = subprocess.check_output(cmd, stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            print(f'curl command failed with error {e.returncode}: {e.output}')
+            continue
+
+        try:
+            result = json.loads(result)
+            return int(result['result']['data'][0]['epoch'])
+        except (KeyError, IndexError, json.JSONDecodeError):
+            print(f'suix_getEpochs rpc request failed: {result}')
+            continue
+    print(f"Failed to get current network epoch after {NUM_RETRIES} tries")
+    exit(1)
+
+
+def get_local_metric(metric: Metric):
+    for i in range(NUM_RETRIES):
+        # try:
+        curl = subprocess.Popen(
+            ['curl', '-s', 'http://localhost:9184/metrics'], stdout=subprocess.PIPE)
+        grep_1 = subprocess.Popen(
+            ['grep', metric.value], stdin=curl.stdout, stdout=subprocess.PIPE)
+        try:
+            result = subprocess.check_output(
+                ['grep', '^[^#;]'], stdin=grep_1.stdout, stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            print(f'curl command failed with error {e.returncode}: {e.output}')
+            continue
+
+        # convert json result to dictionary
+        try:
+            return int(result.split()[1])
+        except (KeyError, IndexError, json.JSONDecodeError):
+            print(
+                f'Failed to get local metric {metric.value}: {result.stdout}')
+            continue
+    print(
+        f"Failed to get local metric {metric.value} after {NUM_RETRIES} tries")
+    exit(1)
+
+
+def main(argv):
+    if len(argv) > 2:
+        print(
+            "Usage: check_synced.py [--end-epoch=END_EPOCH] [--env=ENVIRONMENT]")
+        exit(1)
+
+    opts, args = getopt.getopt(argv, '', ["env=", "end-epoch="])
+
+    env = 'testnet'
+    end_epoch = None
+    for opt, arg in opts:
+        if opt == '--env':
+            env = arg
+        elif opt == '--end-epoch':
+            end_epoch = int(arg)
+
+    if end_epoch is None:
+        end_epoch = get_current_network_epoch(env)
+    print(f'Will attempt to sync to epoch {end_epoch}')
+
+    current_epoch = get_local_metric(Metric.EPOCH)
+    print(f'Current local epoch: {current_epoch}')
+
+    current_checkpoint = get_local_metric(Metric.CHECKPOINT)
+    print(f'Locally highest executed checkpoint: {current_checkpoint}')
+
+    # get local time so that we can measure the time since the last epoch change
+    current_time = datetime.now()
+    while current_epoch < end_epoch:
+        # check that we are making progress
+        time.sleep(10)
```

Comment:
Should this be `CHECKPOINT_STUCK_THRESHOLD_SEC `?
The const is defined but never used in code logic.